src/main/java/ch/uzh/ifi/hase/soprafs22/service/UserService.java

```diff
@@ -531,6 +546,66 @@ public List<User> getMatchedUsers(long userId) {
     return users;
   }
 
+
+  // instantiate demo users
+  public void instantiateDemoUsers() {
+
+    if (areInstantiatedDemoUsers) {
+      // throw exception if demo users are already instantiated
+      throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Demo users are already instantiated");
+
+    } else {
+
+      System.out.println("Instantiating demo users...");
+
+      // ======= create demo users =======
+      User demoUser1 = new User();
+      demoUser1.setUsername("demoUser1");
+      demoUser1.setPassword("demoUser1");
+      demoUser1.setName("Demo User 1");
+      demoUser1.setGender(Gender.MALE);
+      demoUser1 = createUser(demoUser1);
+
+      User demoUser2 = new User();
+      demoUser2.setUsername("demoUser2");
+      demoUser2.setPassword("demoUser2");
+      demoUser2.setName("Demo User 2");
+      demoUser2.setGender(Gender.FEMALE);
+      demoUser2 = createUser(demoUser2);
+
+      User demoUser3 = new User();
+      demoUser3.setUsername("demoUser3");
+      demoUser3.setPassword("demoUser3");
+      demoUser3.setName("Demo User 3");
+      demoUser3.setGender(Gender.OTHER);
+      demoUser3 = createUser(demoUser3);
+
+
+      // ======= create active games =======
+      BlackCard blackCard1 = gameService.getNRandomBlackCards(1).get(0);
+      BlackCard blackCard2 = gameService.getNRandomBlackCards(1).get(0);
+      BlackCard blackCard3 = gameService.getNRandomBlackCards(1).get(0);
+      Game game1 = gameService.createGame(blackCard1, demoUser1.getId());
+      Game game2 = gameService.createGame(blackCard2, demoUser2.getId());
+      Game game3 = gameService.createGame(blackCard3, demoUser3.getId());
+
+
+      // ======= create likes and matches =======
+      Match demoMatch1 = createMatch(demoUser1, demoUser2);
+      setMatch(demoMatch1);
+
+      // TODO: find reason for this
+      // the following code breaks (as far as I can tell between the 2 calls)
+      // Match demoMatch2 = createMatch(demoUser1, demoUser3);
+      // setMatch(demoMatch2);
```

Comment:
https://stackoverflow.com/questions/44650215/jpa-primary-key-violation-when-saving-an-entity-with-many-to-one
I think it might be fixed by saving userId's instead of users, but I'm not sure why it does not work
![image](https://user-images.githubusercontent.com/75019468/166683120-83406d36-4996-4c05-b507-db789a7c466c.png)
Have you created an Issue for this?

Reply:
Also, the error happens in createMatch at saveAndFlush

Reply:
Yes, and this bit of information is also linked in https://github.com/sopra-fs22-group-07/server/issues/166